src/_gettsim/groupings.py

```diff
@@ -230,3 +234,32 @@ def wthh_id_numpy(
             result.append(current_hh_id * 100)
 
     return numpy.asarray(result)
+
+
+def _fail_if_more_than_one_fg_in_hh(
+    hh_id: numpy.ndarray[int],
+    fg_id: numpy.ndarray[int],
+):
+    """
+    Fail if there is more than one `fg_id` in a household.
+
+    Parameters
+    ----------
+    hh_id : numpy.ndarray[int]
+        Array of household IDs.
+    fg_id : numpy.ndarray[int]
+        Array of family group IDs.
+    """
+    unique_hh_ids = numpy.unique(hh_id)
+    error_msg = (
+        "There is at least one household with more than one `fg_id`. GETTSIM does "
+        "not support the endogenous creation of Bedarfsgemeinschaften in this case "
+        "yet. Please provide `bg_id` and `wthh_id` yourself."
+    )
+
+    for idx in unique_hh_ids:
+        # Find all family group IDs for the current household ID
+        fg_ids_in_hh = fg_id[hh_id == idx]
+        # Check if all fg_ids are the same in this household
+        if len(numpy.unique(fg_ids_in_hh)) > 1:
+            raise ValueError(error_msg)
```

Comment:
I think it is okay to fail on the first case here, but it would still be good to provide the faulty `hh_id` to the user in the error message.